### PR TITLE
Swat/feature/security changes default port

### PR DESCRIPTION
**Pull request by sshankar19**

### CODE REVIEW

**Review by jacoblb64 — APPROVED** [review]

File: lib/types/ILifecycleController.d.ts [hunk 1]
```diff
@@ -44,53 +44,53 @@ export interface ILifecycleController {
...
      * @param {PreServerCallChanges} changes - Dictionary of Objects that have been changed from the client
      * @param {CallContext} callContext - Context (number of retries etc)
-     * @param {boolean} [forceUpdate] - Optional parameter passed in. True if this is a retry of the call based on an update conflict. False / undefined otherwise.
-     * 
+     * @param {boolean} forceUpdate -  True if this is a retry of the call based on an update conflict. False / undefined otherwise.
+     * @param {boolean} functionName - The function name as a string
      * @returns {Promise<void>}
      * @memberof ILifecycleController
      */
-    preServerCall?(hasChanges: boolean, changes: PreServerCallChanges, callContext: CallContext, forceUpdate?: boolean): Promise<void>;
```
Comment (nsaenz7) [inline lib/types/ILifecycleController.d.ts:75]: if we're still OK with `forceUpdate` being undefined, would that be the same as it being an optional parameter to pass in?

File: lib/types/ILifecycleController.d.ts [hunk 2]
```diff
@@ -44,53 +44,53 @@ export interface ILifecycleController {
...
+    preServerCall?(hasChanges: boolean, changes: PreServerCallChanges, callContext: CallContext, forceUpdate: undefined | boolean, functionName: string): Promise<void>;
```
Comment (nsaenz7) [inline lib/types/ILifecycleController.d.ts:75]: if we're adding a new, non-optional parameter in the `preServerCall` signature (`functionName`), would that be considered a breaking change for the consuming apps?

File: lib/types/ILifecycleController.d.ts (hunk 1 above)
Comment (sshankar19) [reply lib/types/ILifecycleController.d.ts:75]: This is to get around typescript issues, where you can only have optional parameters at the end of the list of args, although it's technically the same

Comment (nsaenz7) [reply lib/types/ILifecycleController.d.ts:75]: hmm, if typescript expects that to be the case (optional params are always at the end), and most of our other apps are coded that way, I'm wondering if we should stick to that pattern here too...

**Review by nsaenz7 — APPROVED** [review]

**Pull request merged**